marker-api.js

```diff
@@ -1,780 +1,780 @@
+                      if (err) console.log('error renaming file: ' + err);
+
+                      console.log('archive of ' + photo + ' successful');
+                  });
+              });
+
+              // Add to deleted collection
+              var delMarker = new DeletedMarker();
+
+              delMarker.geometry = marker.geometry;
+              delMarker.tags = marker.tags;
+              delMarker.photo_hash = marker.photo_hash;
+              delMarker.user_id = marker.user_id;
+              delMarker.createdDate = marker.createdDate;
+              delMarker.orig_id = marker._id;
+
+              delMarker.save();
+
+              // Delete from db
+              marker.remove();
+
+              res.status(200).json({message: 'marker with id: ' + req.query.marker_id + ' deleted successfully'});
+          });
+      },
+
+      getMarkersByUser: function (req, res) {
+
+              // Make sure user is authorized to see/edit this marker
+              if (!req.user) {
+                  return res.status(403).json({message: 'Unauthorized'});
+              }
+
+              // Update marker data
+              var query = Marker.find({'user_id': req.user._id}).select('approved');
+              
+              query.exec(function (err, markers) {
+                  if (err) {
+                      console.log(err);
+                      return res.status(500).json({message: 'An internal error occurred'});
+                  }
+
+                  if (!markers)
+                      return res.status(204).json({message: 'No markers found.'});
+
+      // convert marker array to object keyed by id
+      var data = {};
+      markers.forEach(function (marker) {
+          data[marker._id] = marker.approved
+      });
+
+                  return res.status(200).json(data);
+              }); // end update
+      },
+
+      getMarkersWithin: function (req, res) {
+
+          // Parse requested coords
+          var bottom_left = [
+              parseFloat(req.query.bottom_left_lng),
+              parseFloat(req.query.bottom_left_lat)
+          ];
+
+          var upper_right = [
+              parseFloat(req.query.upper_right_lng),
+              parseFloat(req.query.upper_right_lat)
+          ];
+
+          
+          // Mongo query
+          var query = Marker.find({
+              geometry: {
+                  $geoWithin: {
+                      $box: [
+                          bottom_left,
+                          upper_right
+                      ]
+                  }
+              }
+          });
+
+          // Paged result if page param exists
+          if (req.query.page) {
+              var res_per_page = 20;
+              var page_index = parseInt(req.query.page) - 1;
+
+              if (page_index >= 0) {
+                  query.skip(page_index * res_per_page);
+              } else {
+                  console.log('invalid page value');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/charlieblevins/dukweb/pullRequest?prid=9134160)
```suggestion
                  console.log("invalid page value");
```